pysat/_constellation.py

```diff
@@ -555,9 +586,16 @@ def load(self, *args, **kwargs):
 
         """
 
+        # Load the data for each instrument
         for instrument in self.instruments:
             instrument.load(*args, **kwargs)
 
+        # Set the year and doy attributes for the constellation and instruments
+        self.yr, self.doy = utils.time.getyrdoy(self.date)
+
+        self._set_inst_attr('yr', self.yr)
```

Comment:
`yr` and `doy` are set on each Instrument as they are loaded. `self.date` maps back to `_index` which requires all instruments to have data, correct?

So if only some instruments have data then doesn't this lead to each instrument having a `yr` and `doy` of None?

Reply:
`self.date` is `const.date` which maps back to the common index.  So I did this but wanted to see what you thought.  As is, if no data is loaded for an instrument, it will have `inst.date=None` and `inst.yr=YEAR; inst.doy=DOY` for the data loaded elsewhere.  Not sure we should be doing it, but thought it would be good to have things reletively consistent across all instruments.  Also totally OK doing away with this if it gives you worries about load/makes you grimice 🦇

Reply:
I think changing the Instrument values should be avoided unless the method is specifically for changing those values. So I do have heartburn about that part. 

I like the addition of yr and doy to the constellation. Also makes sense to assign those directly in the constellation load.

I spent some time writing out a long thing about const.date but that isn't what this pull is about so I took it out instead. It may resurface as an issue.